### ISSUE TITLE
Pass bashly.yml through ERB

### ISSUE THREAD

**Issue by n-rodriguez** [DannyBen/bashly#401]
### Description

So users can generate `allowed` args using some Ruby code. Thank you!

**Comment by DannyBen** [comment]
Good idea. Do you have a sample YAML use case?

**Comment by DannyBen** [comment]
The implementation is ready if you want to try it.
Let me know if it aligned with your use case.

**Comment by n-rodriguez** [comment]
use case:

```yml
commands:
  - name: completions
    help: |-
      Generate bash completions
      Usage: eval "\$(ansible completions)"

  - name: ansible
    help: Ansible related commands
    commands:
      - name: playbook
        help: Run Ansible playbook
        flags:
          - long: --playbook
            short: -p
            arg: playbook
            help: Playbook name
            required: true
            repeatable: false
            allowed:
              <%- Dir['playbooks/**/*.yml'].each do |f| -%>
                <%- _, dir, file = f.split('/') -%>
              - <%= "#{dir}_#{file}".gsub('_', '-').gsub('.yml', '') %>
              <%- end -%>
```

**Comment by n-rodriguez** [comment]
> Let me know if it aligned with your use case.

it does! thank you! ❤️

**Comment by DannyBen** [comment]
Ok cool.

1. Note that I did not use the trim mode '-' - I just used the default ERB settings:

https://github.com/DannyBen/bashly/blob/44294bba54574c0be0c52d96735861d75dbc4351/lib/bashly/extensions/yaml.rb#L8-L10

if there is any reason to load it differently in your opinion, let me know

2. I will release during the weekend
3. As I see you have some DevOps template manipulations to deal with - you might be interested in [Kojo](https://github.com/dannyben/kojo) - I used it a lot to transform Kubernetes manifests into something more human friendly.

**Comment by n-rodriguez** [comment]
> As I see you have some DevOps template manipulations to deal with - you might be interested in [Kojo](https://github.com/dannyben/kojo) - I used it a lot to transform Kubernetes manifests into something more human friendly.

Seems nice, thank you!

> You know you can run `bashly validate --verbose` to see the rendered YAML, yes?

I didn't know, thank you!

**Comment by DannyBen** [comment]
Released 1.0.7 with this feature.
Minimal docs here: https://bashly.dannyb.co/advanced/erb-in-config/

Thanks for the suggestion.

**Comment by n-rodriguez** [comment]
> As I see you have some DevOps template manipulations to deal with - you might be interested in [Kojo](https://github.com/dannyben/kojo) - I used it a lot to transform Kubernetes manifests into something more human friendly.

Actually I did something (a bit) similar but in Crystal : https://github.com/jbox-web/envtpl.cr
